notebooks/problem-sets/quantum_enigma_003.ipynb

```diff
@@ -0,0 +1,395 @@
+    "problem_qc.x(7)\n",
+    "problem_qc.barrier()\n",
+    "problem_qc.x(6)\n",
+    "problem_qc.x(7)\n",
+    "problem_qc.ccx(7,6,0)\n",
+    "problem_qc.x(7)\n",
+    "problem_qc.x(6)\n",
+    "problem_qc.barrier()\n",
+    "\n",
+    "#finding the parity of 1's on squares for which binary numbers finish by 1 and putting the answer on q8\n",
+    "problem_qc.cx(1, 8)\n",
+    "problem_qc.cx(3, 8)\n",
+    "problem_qc.barrier()\n",
+    "\n",
+    "#finding the parity of 1's on squares for which binary numbers have a 1 as second to last digit and putting the answer on q9\n",
+    "problem_qc.cx(2, 9)\n",
+    "problem_qc.cx(3, 9)\n",
+    "problem_qc.barrier()\n",
+    "\n",
+    "problem_qc.draw(output='mpl')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<!-- ::: q-block.exercise -->\n",
+    "\n",
+    "### Quick quiz\n",
+    "\n",
+    "<!-- ::: q-quiz(goal=\"enigma-003-quiz-1\") -->\n",
+    "\n",
+    "<!-- ::: .question -->\n",
+    "\n",
+    "What is the value on $q_6$ after such an operation?\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option(correct) -->\n",
+    "\n",
+    "1. $q_6$ now has the answer to the modulo two addition between $q_4$ and $q_6$.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option -->\n",
+    "\n",
+    "2. An extra qubit would be needed to have the answer to the modulo two addition between $q_4$ and $q_6$.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option -->\n",
+    "\n",
+    "3. No addition has been performed between $q_4$ and $q_6$.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: .option -->\n",
+    "\n",
+    "4. The `CNOT` does not permit to perform modulo two additions.\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: -->\n",
+    "\n",
+    "<!-- ::: -->"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Problem 2\n",
+    "\n",
+    "Can you write the circuit for a 4 by 4 square chess set until you calculate the position of the coin to turn?\n",
+    "\n",
+    "Check your answer by clicking the `Grade` button.\n",
+    "\n",
+    "\n",
+    "<!-- ::: q-block.reminder -->\n",
+    "\n",
+    "### Hints\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 1</summary>\n",
+    "    Start by drawing a 4 by 4 chess board and number each square from 0 to 15 in decimal and binary numbers starting with the top row.\n",
+    "</details>\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 2</summary>\n",
+    "    The trick is now to add (modulo two) all the squares that end with a 1 and to proceed the same way with all squares that have a 1 on their second bit counting from right to left and so on using four extra squares.\n",
+    "</details>\n",
+    "\n",
+    "<details>\n",
+    "    <summary>Hint 3</summary>\n",
+    "    Provide the drawing of Hint 2 with the arrows for the counting\n",
```

Comment:
Before the drawing add:

In the following picture, all squares with green numbers must be added modulo 2 and the answer must be place in the last qubit (q20) of the focus. The same must take place for blue squares with their result on q21, the yellow squares on q22, and the red squares on q23.